wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift

```diff
@@ -79,20 +80,26 @@ final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, Observabl
         return nil
     }
 
+    @MainActor
```

Comment:
@theswiftway  could you explain more in details why it needs to be run on mainThread?

Reply:
As removal of device deals with removing Device from the DB. SyncContext functions do not function as expected when called from a background thread. @netbe